### PR TITLE
Rating component

### PR DESCRIPTION
**Pull request by KeerthanaR29**

### CODE REVIEW

**Review by vczb — COMMENTED** [review]
hello :wave: 

thanks for your PR

please use theme for styling, check other components ( like Button ) to figure out

**Comment by KeerthanaR29** [comment]
Do you want to have primary, secondary and neutral theme like button? or else do you want to add color code to theme.ts? @vczb

**Comment by vczb** [comment]
> Do you want to have primary, secondary and neutral theme like button? or else do you want to add color code to theme.ts? @vczb

you can pick any color from theme, also for sizing use the theme as well, please do not add new colors :)

here you can check all colors that we have

https://github.com/vczb/sagu-ui/blob/main/src/styles/theme/colors.ts

File: src/packages/Rating/style.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  }
+
+  :not(:checked) > label {
+  float: right;
+  width: 1em;
+  overflow: hidden;
+  white-space: nowrap;
+  cursor: pointer;
+  font-size: 30px;
+  color: #ccc;
```
Comment (vczb) [inline src/packages/Rating/style.ts:17]: image
please, take a look on this component to undertand how to use theme

https://github.com/vczb/sagu-ui/blob/main/src/packages/Alert/styles.ts

File: src/styles/theme/colors.ts [hunk 2]
```diff
@@ -10,6 +10,12 @@ export const base = {
   success: '#4caf50'
 }
 
+export const rating = {
```
Comment (vczb) [inline src/styles/theme/colors.ts]: hello, let's use some existing color :)

File: src/styles/theme/index.ts [hunk 3]
```diff
@@ -46,6 +46,9 @@ export default {
     xlarge: '4.8rem',
     xxlarge: '5.6rem'
   },
+  width: {
```
Comment (vczb) [inline src/styles/theme/index.ts]: pick the sizing from here 

https://github.com/vczb/sagu-ui/blob/b0db230d9c5c919d80452df93ed4208ca7f31de6/src/styles/theme/index.ts#L37-L47

File: src/packages/Rating/index.tsx [hunk 4]
```diff
@@ -16,6 +16,9 @@ const Rating = ({ name, value, onChange }: RatingProps) => {
   }
   // eslint-disable-next-line @typescript-eslint/no-empty-function
   useEffect(() => {}, [selectedValue])
+  useEffect(() => {
```
Comment (vczb) [inline src/packages/Rating/index.tsx:19]: seems like you have declared useEffect twice, btw one of those is doing nothing...

image

**Review by vczb — COMMENTED** [review]
hey! thanks for your PR, I left some comments for adjustments :) let me know when you are done or if you have any question

**Comment by vczb** [comment]
JIC you are facing linter issues you can run `yarn prettier:format`

File: src/packages/Rating/index.tsx (hunk 4 above)
Comment (KeerthanaR29) [reply src/packages/Rating/index.tsx:19]: the unused useEffect is to rerender the component when we are clicking on one of the stars in Rating component

**Review by vczb — APPROVED** [review]
LGTM

**Pull request merged**